### DISCUSSION TITLE
wasmGC

### DISCUSSION THREAD

**Discussion by mtt-artis** [leptos-rs/leptos#1116] [General]
Hi!
leptos is my favorite rust frontend framework  so far! 
thank you to the community for the time you spent on it!
i juste watch  "WebAssembly: A new development paradigm for the web" by  Google Chrome Developers
on youtube https://www.youtube.com/watch?v=RcHER-3gFXI&t=0s.

how wasmGC will impact leptos in termsof performance and bundle size ?

**Comment by gbj** [comment]
AFAIK it will have no impact on Leptos, or any Rust/WASM framework, except increasing the competition from other languages. As I understand it the point of WASM GC is to allow garbage-collected languages to compile to WASM without having to ship their own garbage collector as part of the WASM binary. Rust is not a garbage-collected language so this is mostly irrelevant. 

Unless I'm mistaken about what the WASM GC spec is for.

**Comment by mtt-artis** [comment]
I get it for the garbage collector but if you watch this slide https://www.youtube.com/watch?v=RcHER-3gFXI&t=15m50s it also talks about direct access to the dom, canvas, webgpu and webgl. 
This will reduce glue code. I am wrong?
